app/services/form1010cg/service.rb

```diff
@@ -83,38 +80,43 @@ def build_metadata
     # The result will be cached and subsequent calls will return the cached value, preventing additional api requests.
     #
     # @param form_subject [String] The key in the claim's data that contains this person's info (ex: "veteran")
-    # @return [String | NOT_FOUND] Returns `true` if the form subject is a veteran and NOT_CONFIRMED otherwise.
+    # @return [String | NOT_FOUND] Returns the icn of the form subject if found, and NOT_FOUND otherwise.
     def icn_for(form_subject)
       cached_icn = @cache[:icns][form_subject]
       return cached_icn unless cached_icn.nil?
 
-      begin
-        response = mvi_service.find_profile(build_user_identity_for(form_subject))
-      rescue MVI::Errors::RecordNotFound
+      response = mvi_service.find_profile(build_user_identity_for(form_subject))
+
+      case response.status
+      when 'OK'
+        return @cache[:icns][form_subject] = response.profile.icn
```

Comment:
Noticed that you removed the safe navigation operators response&.profile&.icn, would response.profile ever be nil?

Reply:
Since we're doing a search for a profile, and the status is "OK", there should be a profile there. Also had a safeguard on ICN, but found out that an ICN should always be present on an MVI profile.

Reply:
Ok great, looks good